### ISSUE TITLE
[BUG] Improve Image Download Handling

### ISSUE THREAD

**Issue by WebberHuang1118** [harvester/harvester#5044]
**Describe the bug**
Downloaded images can't be uploaded again

**To Reproduce**
1. Creating a virtual machine image with upload from local
2. Downloading the image just uploaded
3. Creating another virtual image by uploading the newly downloaded image
4. Virtual image creation failed with `Upload failed: the uploaded file size 665368738 should be a multiple of 512 bytes since Longhorn uses directIO by default`

**Expected behavior**
Successfully uploading an image which is originally downloaded from Harvester 

**Environment**
 - Harvester ISO version: Harvester master-branch (commit `0861f5be4533418bbe148efc5f0bc32f07fe4220`)
 
 **Additional context**
After investigation, we found out LH changes the behavior to [compress the downloaded image](https://github.com/longhorn/backing-image-manager/pull/153), in the current Harvester implementation, the downloaded image file will be named with `.img` suffix by default, however, this file is compressed by LH, if users want to upload the downloaded image, it should be decompressed first.

Harvester should change the [attachment in the request header as LH](https://github.com/longhorn/backing-image-manager/blob/28b787c0ddfc95fb6af5573294e6366cdd0f87f2/pkg/sync/service.go#L246-L251) to name the downloaded image with `.gz` suffix, which could avoid confusing, and a document updating is also required.

**Workaround**
Rename the downloaded image as `.gz` suffix, and decompress it with `$ gzip -d <image name>`, then the image file should be eligible to upload.

**Comment by irishgordo** [comment]
Hi, @WebberHuang1118 I'm noticing some issues with this still, but not sure if due to underlying configuration.

**Validation Failed:**

I'm still noticing the issue is present on:
- Harvester ISO Version: v1.3.0-rc1
- Rancher Dashboard Version: `commit c3ae9bc14ad1a306025c0882eefde7baef3d046d` ( containing `commit 65366810efe83db05512194589771b8c3a0a0914` ) (has fix for volumes)
- Underlying Infrastructure: qemu/kvm

![Screenshot from 2024-02-01 16-27-53](https://github.com/harvester/harvester/assets/5370752/8747744f-dfd3-4c34-aea6-64803690686c)

**Comment by WebberHuang1118** [comment]
Hi @irishgordo,
The related commit is [12d5e49354cdbdd9da88ffe46d452e55217cfee8](https://github.com/harvester/harvester/commit/12d5e49354cdbdd9da88ffe46d452e55217cfee8), which is not included in [1.3.0-rc1 history](https://github.com/harvester/harvester/commits/v1.3.0-rc1) (because it's merged after we cut 1.3.0-rc1).

Can you verify with the master branch? Thanks.

**Comment by irishgordo** [comment]
Hi @WebberHuang1118  - thanks for that mention :smile: 
Retested on:
```
NAME="SLE Micro"
VERSION="5.4"
VERSION_ID="5.4"
PRETTY_NAME="Harvester master"
ID="sle-micro-rancher"
ID_LIKE="suse"
ANSI_COLOR="0;32"
CPE_NAME="cpe:/o:suse:sle-micro-rancher:5.4"
VARIANT="Harvester"
VARIANT_ID="Harvester-20240202-2d32319"
GRUB_ENTRY_NAME="Harvester master"
```
(  Version: master-a590f589-head  )

And this looks great :smile: :+1: 
I'll go ahead and close this out
![Screenshot from 2024-02-02 15-48-38](https://github.com/harvester/harvester/assets/5370752/0d503434-e7b4-432f-a429-2659225b0e90)
![Screenshot from 2024-02-02 15-47-22](https://github.com/harvester/harvester/assets/5370752/16023cf8-1cdd-473e-afcc-b488952dec1f)
![Screenshot from 2024-02-02 15-45-50](https://github.com/harvester/harvester/assets/5370752/6a835471-fd08-448c-9104-2936ea72465d)
